lib/faraday_middleware/response/caching.rb

```diff
@@ -86,10 +86,9 @@ def full_key?
       @full_key ||= @options[:full_key]
     end
 
-    def custom_status_code
-      Array(@options[:cacheable_status_code]).map do |status_code|
-        CACHEABLE_STATUS_CODES.include?(status_code.to_i) ? status_code.to_i : nil
-      end.compact
+    def custom_status_codes
+      @custom_status_codes ||= CACHEABLE_STATUS_CODES & Array(@options[:cacheable_status_code]).map(&:to_i)
+      @custom_status_codes.any? ? @custom_status_codes : CACHEABLE_STATUS_CODES
```

Comment:
Would this do the same thing?

```suggestion
      @custom_status_codes ||= begin
        codes = CACHEABLE_STATUS_CODES & Array(@options[:cacheable_status_code]).map(&:to_i)
        codes.any? ? codes : CACHEABLE_STATUS_CODES
      end
```

Reply:
that works as well! i didnt know you could memoized an entire block... this is neat! will make the change!